src/pages/workspace/accounting/PolicyAccountingPage.tsx

```diff
@@ -52,9 +112,15 @@ function PolicyAccountingPage({policy, connectionSyncProgress}: PolicyAccounting
     const [threeDotsMenuPosition, setThreeDotsMenuPosition] = useState<AnchorPosition>({horizontal: 0, vertical: 0});
     const [isDisconnectModalOpen, setIsDisconnectModalOpen] = useState(false);
     const threeDotsMenuContainerRef = useRef<View>(null);
+
     const isSyncInProgress = !!connectionSyncProgress?.stageInProgress && connectionSyncProgress.stageInProgress !== CONST.POLICY.CONNECTIONS.SYNC_STAGE_NAME.JOB_DONE;
-    const policyIsConnectedToAccountingSystem = Object.keys(policy.connections ?? {}).length > 0;
-    const policyID = policy.id ?? '';
+    const connectedIntegration = accountingIntegrations.find((integration) => !!policy?.connections?.[integration]) ?? connectionSyncProgress?.connectionName;
+    const policyID = policy?.id ?? '';
+
+    const policyConnectedToXero = connectedIntegration === CONST.POLICY.CONNECTIONS.NAME.XERO;
+
+    const tenants = useMemo<Tenant[]>(() => policy?.connections?.xero?.data?.tenants ?? [], [policy?.connections?.xero.data.tenants]);
```

Comment:
@filip-solecki @cdOut I'm getting a crash on connecting Xero for the first time.

`Uncaught TypeError: Cannot read properties of of undefined (reading 'tenants') at PolicyAccountingPage (PolicyAccountingPage.tsx:114:4)` 


https://github.com/Expensify/App/assets/29673073/c69ca2c6-417c-4749-bd83-30adb6f86c80